packages/logger-plugin/src/log-writer.ts

```diff
@@ -0,0 +1,62 @@
+import { NgxsLoggerPluginOptions } from './symbols';
+export class LogWriter {
+  private logger: any;
+
+  constructor(private options: NgxsLoggerPluginOptions) {
+    this.options = this.options || <any>{};
+    this.logger = options.logger || console;
+  }
+
+  startGroup(message: string) {
+    const startGroupFn = this.options.collapsed
+      ? this.logger.groupCollapsed
+      : this.logger.group;
+    try {
+      startGroupFn.call(this.logger, message);
+    } catch (e) {
+      console.log(message);
+    }
+  }
+
+  endGroup() {
+    try {
+      this.logger.groupEnd();
+    } catch (e) {
+      this.logger.log('—— log end ——');
+    }
+  }
+
+  logGrey(title: string, payload: any) {
+    this.log(title, 'color: #9E9E9E; font-weight: bold', payload);
```

Comment:
maybe we can

```ts
export enum STYLE {
 GREY = 'color: #9E9E9E; font-weight: bold',
 GREEN = 'color: #4CAF50; font-weight: bold',
 REDIS = 'color: #FD8182; font-weight: bold'
}
```

Reply:
They are only used in this one place so the name of the colour is obvious.
Are you aware of the amount of generated code that these enums add? It is quite a bit.
I may just extract some variables to make it clearer though.

Reply:
`I may just extract some variables to make it clearer though` - please

Reply:
I added variables to be more descriptive of the style.